### PR TITLE
Complete Unit Tests for OOP School Library - Project 7

### PR DESCRIPTION
**Pull request by CVILLA09**
### Overview  
This pull request fulfills the requirements for "Project 7: unit tests" in the OOP School Library project. The focus is on adding comprehensive unit tests for all classes in the codebase, ensuring each method is accurately tested.

### Changes Made  
- Added unit tests for the following classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - `BaseDecorator`
  - `TrimmerDecorator`
  - `CapitalizeDecorator`
- Followed Ruby's best practices for writing tests using RSpec.
- Ensured no linter errors.

### Ruby Files  
- Added new RSpec files under the `spec/models` directory for each class being tested.
- These include `book_spec.rb`, `person_spec.rb`, `student_spec.rb`, `teacher_spec.rb`, `classroom_spec.rb`, `rental_spec.rb`, `base_decorator_spec.rb`, `trimmer_decorator_spec.rb`, and `capitalize_decorator_spec.rb`.

### Conclusion  
This pull request successfully completes the unit testing requirements for Project 7 in the OOP School Library project. All classes and methods have been thoroughly tested, ensuring robustness and adherence to Ruby's best practices.

### CODE REVIEW

File: spec/models/person_spec.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require './person'
+require './book'
+require './rental'
+
+RSpec.describe Person do
```
Comment (zeff96) [inline spec/models/person_spec.rb:5]: - [ ] Hello there! 🌟 In order to guarantee that every method within the "Person" class is thoroughly tested, could you please add a test case that specifically covers the "add_rental" method? This test will ensure that the "add_rental" method is functioning correctly and that it's included in the test suite for comprehensive test coverage. 🧪👨‍🔬 Thank you! 😊

File: spec/models/base_decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+require_relative '../../base_decorator'
```
Comment (zeff96) [inline spec/models/base_decorator_spec.rb]: - [ ] Hey there! 🚀 I've observed an issue with the relative path you're using. It appears you're referencing "base_decorator" at the root of your project, but the actual path should be "decorator." 😬 Kindly double-check this and make the necessary correction to ensure everything works smoothly. 🛠️ Your attention to this matter is greatly appreciated! 🙏

```suggestion
require_relative '../../decorator'
```

image

**Review by zeff96 — CHANGES_REQUESTED** [review]
### Status: CHANGES REQUIRED ♻️
Hi @CVILLA09 and @Felipe-Perez-Ferraro , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
✔️ Linters passing
✔️ Correct GitFlow

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Danieldotcomcoder — APPROVED** [review]
Hello @CVILLA09 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it 👍 👍 😄 👏 

Congratulations! 🎉 🎉 🎉 

## Highlights
- All Tests are passing✔✔
- Clear and nice code ✔✔

comp-2
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**